torch/_meta_registrations.py

```diff
@@ -1985,7 +1998,10 @@ def activate_meta():
         }:
             pass
         else:
-            _meta_lib_dont_use_me_use_register_meta.impl(op_overload, fn)
+            if op_overload.name().startswith("c10d"):
+                _meta_c10d_lib_dont_use_me_use_register_meta.impl(op_overload, fn)
+            else:
+                _meta_lib_dont_use_me_use_register_meta.impl(op_overload, fn)
```

Comment:
What would this look like without the hack?

Reply:
this line won't change.  the definition of `_meta_lib_dont_use_me_use_register_meta` would switch from using "AutogradMeta" to  "Meta".